clang-tools-extra/test/clang-tidy/checkers/bsl-function-noexcept.cpp

```diff
@@ -0,0 +1,83 @@
+// RUN: %check_clang_tidy %s bsl-function-noexcept %t
+
+#include <stdexcept>
+
+void f1();
+void f2() noexcept;
+void f3() noexcept(true);
+void f4() noexcept(false);
+void f5() noexcept {
+	try 
+	{
+		f1();
+	}
+	catch (std::exception& e) {}
+
+	f2();
+	f3();
+
+	try
+	{
+		f4();
+	}
+	catch (std::exception& e) {}
+}
+
+template <class T>
+void f6() noexcept(noexcept(T()));
+
+template <class T>
+class A
+{
+public:
+	A() noexcept(noexcept(T())) {}		// Non-compliant?
```

Comment:
compliant, because you are using the noexcept specifier. The use of the noexcept operator does not change this.